include/beman/execution26/detail/intrusive_queue.hpp

```diff
@@ -0,0 +1,59 @@
+// include/beman/execution26/detail/intrusive_queue.hpp               -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef INCLUDED_BEMAN_EXECUTION26_DETAIL_INTRUSIVE_QUEUE
+#define INCLUDED_BEMAN_EXECUTION26_DETAIL_INTRUSIVE_QUEUE
+
+#include <cassert>
+#include <utility>
+
+namespace beman::execution26::detail {
+
+template <auto Next>
+class intrusive_queue;
+
+//! @brief  This data structure is an intrusive queue that is not thread-safe.
+template <class Item, Item* Item::*Next>
+class intrusive_queue<Next> {
+  public:
+    ~intrusive_queue() { assert(!head_); }
+
+    //! @brief  Pushes an item to the queue.
+    auto push(Item* item) noexcept -> void {
```

Comment:
It seems there should either be a precondition `assert(not item->next)` (if the operation pushes just one element) or before updating `last` the list of items hanging off `item` needs to be walked.

Reply:
Yes I was lazy and introduced bugs. Whatever I pop in atomic_intrusive_queue is a singly linked list that I want to traverse. I will revise this